src/service/chat.js

```diff
@@ -143,9 +143,11 @@ const sendSignedDMOtherProfile = async ({user_id, message}) => {
   return Promise.reject(response.data?.data);
 };
 
-const getOrCreateAnonymousChannel = async (userId) => {
+const getOrCreateAnonymousChannel = async (userId, oldChannelId = null, context = null) => {
   const payload = {
-    members: [userId]
+    members: [userId],
+    oldChannelId,
+    context
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `getOrCreateAnonymousChannel` now accepts two additional parameters: `oldChannelId` and `context`. However, there's no validation or error handling for these new parameters. It's important to ensure that the values passed are of the expected type and format to prevent potential issues down the line.

```diff
const getOrCreateAnonymousChannel = async (userId, oldChannelId = null, context = null) => {
   if (typeof oldChannelId !== 'string' && oldChannelId !== null) {
       throw new Error('Invalid oldChannelId');
   }
   if (typeof context !== 'object' && context !== null) {
       throw new Error('Invalid context');
   }
   const payload = {
    members: [userId],
    oldChannelId,
    context
   };
 
   try {
```
This will help catch any errors early and make debugging easier.


<!-- This is an auto-generated comment by OSS CodeRabbit -->